perl/0006-perl-5.36.0-msys2.patch

```diff
@@ -2241,23 +415,54 @@ index 4715e9d..5da0af0 100644
          qw(PATH LD_LIBRARY_PATH LANG PERL_BADLANG SHELL HOME LOGDIR LANGUAGE);
      push @env, $Config{ldlibpthname} if $Config{ldlibpthname} ne '';
 -    push @env, grep /^(?:PERL|LC_|LANG|CYGWIN)/, keys %ENV;
-+    push @env, grep /^(?:PERL|LC_|LANG|MSYS)/, keys %ENV;
++    push @env, grep /^(?:PERL|LC_|LANG|CYGWIN|MSYS)/, keys %ENV;
      my %env;
      @env{@env} = @env;
      for my $env (sort keys %env) {
-diff --git a/win32/FindExt.pm b/win32/FindExt.pm
-index 4d46821..210047e 100644
---- a/win32/FindExt.pm
-+++ b/win32/FindExt.pm
-@@ -24,7 +24,7 @@ sub apply_config {
-       unless ($config->{i_dbm} || $config->{i_rpcsvcdbm}) && !$config->{d_cplusplus};
-     push @no, "Amiga.*" unless $^O eq "amigaos";
-     push @no, "VMS.*" unless $^O eq "VMS";
--    push @no, "Win32.*" unless $^O eq "MSWin32" || $^O eq "cygwin";
-+    push @no, "Win32.*" unless $^O eq "MSWin32" || $^O eq "cygwin" || $^O eq "msys";
- 
-     $no = join('|', @no);
-     $no = qr/^(?:$no)$/i;
--- 
-2.37.1
-
+--- perl-5.38.2/hints/cygwin.sh.orig	2025-02-01 13:19:09.488368500 +0100
++++ perl-5.38.2/hints/cygwin.sh	2025-02-01 13:20:42.942658200 +0100
+@@ -19,15 +19,15 @@
+     plibpth=`cd $plibpth && pwd`
+ fi
+ so='dll'
+-# - eliminate -lc, implied by gcc and a symlink to libcygwin.a
++# - eliminate -lc, implied by gcc and a symlink to libmsys-2.0.a
+ libswanted=`echo " $libswanted " | sed -e 's/ c / /g'`
+-# - eliminate -lm, symlink to libcygwin.a
++# - eliminate -lm, symlink to libmsys-2.0.a
+ libswanted=`echo " $libswanted " | sed -e 's/ m / /g'`
+-# - eliminate -lutil, symbols are all in libcygwin.a
++# - eliminate -lutil, symbols are all in libmsys-2.0.a
```

Comment:
Is it really worth patching comments?

Reply:
Fair point, but I just copied over the `hints/msys.sh` changes to `hints/cygwin.sh` (and adjusting the arch strings) to avoid any other side effects of this PR.

Reply:
so cygwin.sh probably diverged since msys.sh was created

Reply:
yeah, likely: https://github.com/Perl/perl5/blame/blead/hints/cygwin.sh